packages/shared/src/components/Feed.tsx

```diff
@@ -283,25 +284,42 @@ export default function Feed<T>({
       return item.post.id === openedPost;
     });
 
-    if (index === items.length) {
-      await fetchPage();
+    if (index === items.length - 1) {
+      if (isFetchingNextPage) {
+        return null;
+      }
+
+      setIsFetchingNextPage(true);
+      return fetchPage();
     }
 
+    if (items[index + 1].type === 'placeholder') {
+      return null;
+    }
+
+    setIsFetchingNextPage(false);
     const offset = items[index + 1].type !== 'post' ? 2 : 1;
     const item = items[index + offset];
 
     if (item?.type !== 'post') {
-      return;
+      return null;
     }
 
     trackEvent({
       origin: 'article modal',
       event_name: 'navigate next',
       target_id: item.post.id,
     });
-    setOpenedPost(item.post.id);
+
+    return setOpenedPost(item.post.id);
   };
 
+  useEffect(() => {
+    if (openedPost && isFetchingNextPage) {
+      onNext();
+    }
+  }, [items, openedPost, isFetchingNextPage]);
```

Comment:
Why do we need this?

Reply:
Because we will have to wait for the items to get updated when you reach the end of the list. I tried doing async in a single function and re-calling the `onNext` in itself but it was still referencing the same object (old list). So I had to ensure the object was updated already before moving forward.